src/onnx/parse_where.cpp

```diff
@@ -40,28 +40,41 @@ struct parse_where : op_parser<parse_where>
                           const onnx_parser::node_info& info,
                           std::vector<instruction_ref> args) const
     {
-        auto lens =
-            compute_broadcasted_lens(args[0]->get_shape().lens(), args[1]->get_shape().lens());
-        lens = compute_broadcasted_lens(lens, args[2]->get_shape().lens());
-        if(args[0]->get_shape().lens() != lens)
+        // This op. doesn't currently support a mix of static and dynamic shapes.
+        // Also, TODO: broadcasting for dynamic shapes isn't implemented at time of writing.
+        //   When it becomes available, add broadcasting steps in the dynamic shape case.
+        // For now for dynamic shapes, just insert the Where op.  All shapes must be the
+        // same for it to succeed.
```

Comment:
We do support broadcasting dynamic shapes

Reply:
comment updated, commit 9b43b0499a